app/models/infra_conversion_job.rb

```diff
@@ -0,0 +1,167 @@
+    message = prep_message("Preflight check has failed: #{error}")
+    _log.info(message)
+    abort_conversion(message, 'error')
+  end
+
+  def abort_conversion(message, status)
+    migration_task.cancel
+    queue_signal(:abort_job, message, status)
+  end
+
+  def polling_timeout(poll_type)
+    count = "#{poll_type.to_s}_count".to_sym
+    max = "#{poll_type.to_s}_max".to_sym
+    context[count] = (context[count] || 0) + 1
+    context[count] > options[max]
+  end
+
+  def poll_conversion
+    return abort_conversion("Polling times out", 'error') if polling_timeout(:poll_conversion)
+
+    message = "Getting conversion state"
+    _log.info(prep_message(message))
+
+    if migration_task.options[:virtv2v_wrapper].nil? || migration_task.options[:virtv2v_wrapper]['state_file'].nil?
+      message = "options[:virtv2v_wrapper]['state_file'] not available, continuing poll_conversion"
+      _log.info(prep_message(message))
+      return queue_signal(:poll_conversion, :deliver_on => Time.now.utc + options[:conversion_polling_interval])
+    end
+
+    begin
+      migration_task.get_conversion_state # migration_task.options will be updated
+    rescue => exception
+      _log.log_backtrace(exception)
+      return abort_conversion("Conversion error: #{exception}", 'error')
+    end
+
+    v2v_status = migration_task.options[:virtv2v_status]
+    message = "virtv2v_status=#{status}"
+    _log.info(prep_message(message))
+
+    case v2v_status
+    when 'active'
+      queue_signal(:poll_conversion, :deliver_on => Time.now.utc + options[:conversion_polling_interval])
+    when 'failed'
+      message = "disk conversion failed"
+      abort_conversion(prep_message(message), 'error')
+    when 'succeeded'
+      message = "disk conversion succeeded"
+      _log.info(prep_message(message))
+      queue_signal(:start_post_stage)
+    else
+      message = prep_message("Unknown converstion status: #{v2v_status}")
+      abort_conversion(message, 'error')
+    end
+  end
+
+  def start_post_stage
+    # once we refactor Automate's PostTransformation into a job, we kick start it here
+    message = "To start polling for PostTransformation progress"
+    _log.info(prep_message(message))
+    queue_signal(:poll_post_stage, :deliver_on => Time.now.utc + options[:conversion_polling_interval])
+  end
+
+  def poll_post_stage
+
+    return abort_conversion("Polling times out", 'error') if polling_timeout(:poll_post_stage)
+
+    message = "PostTransformation state=#{migration_task.state}, status=#{migration_task.status}"
+    _log.info(prep_message(message))
+    if migration_task.state == 'finished'
+      self.status = migration_task.status
+      queue_signal(:finish)
+    else
+      queue_signal(:poll_post_stage, :deliver_on => Time.now.utc + options[:conversion_polling_interval])
+    end
+  end
+
+  def queue_signal(*args, deliver_on: nil)
+    role     = options[:role] || "ems_operations"
+    priority = options[:priority] || MiqQueue::NORMAL_PRIORITY
```

Comment:
Are the role and/or priority being set anywhere where this is called?  Could we drop the `||` and just set it to ems_ops and normal priority?